addons/binding/org.openhab.binding.xiaomivacuum/src/main/java/org/openhab/binding/xiaomivacuum/handler/XiaomiMiIoHandler.java

```diff
@@ -0,0 +1,338 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * The {@link XiaomiMiIoHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Marcel Verpaalen - Initial contribution
+ */
+public abstract class XiaomiMiIoHandler extends BaseThingHandler {
+    private final Logger logger = LoggerFactory.getLogger(XiaomiMiIoHandler.class);
+
+    protected ScheduledFuture<?> pollingJob;
+    protected XiaomiVacuumBindingConfiguration configuration;
+    protected MiIoDevices miDevice = MiIoDevices.UNKNOWN;
+
+    protected JsonParser parser;
+    protected byte[] token;
+
+    protected RoboCommunication roboCom;
+    protected int lastId;
+
+    protected ExpiringCache<String> network;
+
+    protected final long CACHE_EXPIRY = TimeUnit.SECONDS.toMillis(5);
+
+    public XiaomiMiIoHandler(Thing thing) {
+        super(thing);
+        parser = new JsonParser();
+    }
+
+    @Override
+    public abstract void handleCommand(ChannelUID channelUID, Command command);
+
+    @Override
+    public void initialize() {
+        logger.debug("Initializing Mi IO device handler '{}' with thingType {}", getThing().getUID(),
+                getThing().getThingTypeUID());
+        configuration = getConfigAs(XiaomiVacuumBindingConfiguration.class);
+        if (!tolkenCheckPass(configuration.token)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Token required. Configure token");
+            return;
+        }
+        scheduler.schedule(this::initializeData, 0, TimeUnit.SECONDS);
+        scheduler.schedule(this::getDeviceType, 1, TimeUnit.SECONDS);
+        int pollingPeriod = configuration.refreshInterval;
+        if (pollingPeriod > 0) {
+            pollingJob = scheduler.scheduleWithFixedDelay(this::updateData, 2, pollingPeriod, TimeUnit.SECONDS);
+            logger.debug("Polling job scheduled to run every {} sec. for '{}'", pollingPeriod, getThing().getUID());
+        } else {
+            logger.debug("Polling job disabled. for '{}'", getThing().getUID());
+        }
+    }
+
+    private boolean tolkenCheckPass(String tokenSting) {
+        boolean tokenPassed = true;
+        switch (tokenSting.length()) {
+            case 32:
+                if (tokenSting.equals("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")
+                        || tokenSting.equals("00000000000000000000000000000000")) {
+                    tokenPassed = false;
+                } else {
+                    token = Utils.hexStringToByteArray(tokenSting);
+                }
+                break;
+            case 16:
+                token = tokenSting.getBytes();
+                break;
+            default:
+                tokenPassed = false;
+        }
+        return tokenPassed;
```

Comment:
why not simply return true here and return false in all cases, where tokenPassed is set to false? :)